aws-android-sdk-kinesisvideo/src/main/java/com/amazonaws/kinesisvideo/http/DistinguishedNameParser.java

```diff
@@ -0,0 +1,392 @@
+package com.amazonaws.kinesisvideo.http;
+
+import javax.security.auth.x500.X500Principal;
+
+/**
+ * A distinguished name (DN) parser. This parser only supports extracting a string value from a DN.
+ * It doesn't support values in the hex-string style.
+ *
+ * Adapted from https://github.com/square/okhttp/blob/okhttp_3.9.x/okhttp/src/main/java/okhttp3/internal/tls/DistinguishedNameParser.java
```

Comment:
Did you consider adding a dependency on okhttp instead of copying this in?

Reply:
@richardmcclellan I did try to add it to the dependency list, but android blocked it from being imported. But, probably that was a different version, let me try it again to confirm this.

Reply:
@richardmcclellan it looks like it didn't work because it's an internal package. It doesn't seem that OkHTTPClient exposes this as an external API.

Reply:
Got it.  A better solution would probably be to go all in and just use `OkHttp` instead of `Apache`, right?     With `OkHttp`, hostname verification is [enabled by default](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/-builder/ssl-socket-factory/).

Reply:
Discussed offline - Migrating to `OkHttp` is ideal long term, but for now we are going with this solution.